src/main/java/org/wildfly/security/auth/server/MechanismConfiguration.java

```diff
@@ -41,14 +43,19 @@
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
  */
 public final class MechanismConfiguration {
+
+    private final MechanismConfiguration parent;
```

Comment:
I'm not sure I am comfortable with the added complexity of an arbitrary hierarchy of configurations.  What use case does an open-ended hierarchy allow that having a default plus per-host configurations does not?